extensions/applicationinsights-clickanalytics-js/Tests/Unit/src/ClickEventTest.ts

```diff
@@ -29,6 +30,138 @@ export class ClickEventTest extends AITestClass {
+                    requestUri: "",
+                    pageName: "",
+                    pageType: ""
+                };
+
+                const clickAnalyticsPlugin = new ClickAnalyticsPlugin();
+                const core = new AppInsightsCore();
+                const channel = new ChannelPlugin();
+               
+                core.initialize({
+                    instrumentationKey: "testIkey",
+                    extensionConfig : {
+                    }
+                } as IConfig & IConfiguration, [clickAnalyticsPlugin, channel]);
+                this.onDone(() => {
+                    core.unload(false);
+                });
+
+                core.config["extensionConfig"] =  core.config["extensionConfig"]?  core.config["extensionConfig"] : {};
+                let extConfig = core.config["extensionConfig"][clickAnalyticsPlugin.identifier];
+                Assert.ok(extConfig, "default config should not be null");
+
+                //check defaults
+                Assert.equal(extConfig.autoCapture,true, "autoCapture should be true by default");
+                Assert.deepEqual(extConfig.pageTags, {}, "pageTags should be {} by default");
+                Assert.equal(extConfig.defaultRightClickBhvr, "", "defaultRightClickBhvr should be '' by default");
+                Assert.equal(extConfig.dropInvalidEvents, false, "dropInvalidEvents should be false by default");
+                Assert.equal(extConfig.urlCollectHash, false, "urlCollectHash should be false by default");
+                Assert.equal(extConfig.urlCollectQuery, false, "urlCollectQuery should be false by default");
+                Assert.deepEqual(extConfig.dataTags, dataTagsDefault, "udataTags should be set by default");
+                Assert.deepEqual(extConfig.coreData, coreDataDefault, "udataTags should be set by default");
+
+                Assert.ok(extConfig.callback, "callback should be set by default");
+                let callbacks = extConfig.callback;
+                let pageActionPageTags = callbacks.pageActionPageTags;
+                let pageName = callbacks.pageName;
+                let contentName = callbacks.contentName;
+                Assert.equal(pageName(), null, "pageName should be set by default");
+                Assert.equal(pageActionPageTags(), null, "pageActionPageTags should be set by default");
+                Assert.equal(contentName(), null, "contentName should be set by default");
+                
+                Assert.ok(extConfig.behaviorValidator,"behaviorValidator should be set by default");
+                let bhvVal = extConfig.behaviorValidator;
+                Assert.equal(bhvVal(), "", "behaviorValidator should return ''");
+                Assert.equal(bhvVal(1), 1, "behaviorValidator should return key");
+
+                //config(non-Object) changes
+                core.config["extensionConfig"][clickAnalyticsPlugin.identifier].urlCollectHash = true;
```

Comment:
Suggestion: Just to make this a little easier to read you can assign `core.config["extensionConfig"][clickAnalyticsPlugin.identifier]` to a local variable